### ISSUE TITLE
Thrown errors are swallowed after using find().exec()

### ISSUE THREAD

**Issue by seppevs** [Automattic/mongoose#4633]
Whenever .find().exec() is invoked on a Mongoose Model, all thrown errors will be swallowed.

Here is a unit test which demonstrates this issue:

````javascript
'use strict';

const assert = require('assert')
const mongoose = require('mongoose')
const Schema = mongoose.Schema
  , ObjectId = Schema.ObjectId;

describe('errors swallowed', function () {

  const BlogPostSchema = new Schema({
    title: String
    , body: String
    , date: Date
  });

  const BlogPost = mongoose.model('BlogPost', BlogPostSchema);

  before(() => {
    mongoose.connect("mongodb://localhost/some_db");
  })

  it('test 1', function (done) {

    BlogPost.find({}).exec(function () {
      console.log("this will print 0")
      assert.equal(1, 1)
      console.log("0 ... no problem");
      done();
    });
  });

  it('test 2', function (done) {
    BlogPost.find({}).exec(function () {
      console.log("this will print 1")
      assert.equal(1, 2)
      console.log("this will NOT print")
      done();
    })
  });

  it('test 3', function (done) {
    BlogPost.find({}).exec(function () {
      console.log("this will print 2")
      throw new Error("this error is swallowed");
      console.log("this will NOT print")
      done();
    });
  });

});
````

This issue occurs on:
* mongoose v4.5.10 (Node v4.4.4)
* mongoose v4.6.4 (Node v4.4.4)

**Comment by seppevs** [comment]
I have closed this issue because it did not occur with the latest mongoose version (v.4.6.4)

**Comment by vkarpov15** [comment]
Yeah this was fixed in 4.6.0. Glad upgrading fixed your issue :+1:
